extmod/nimble: Update to NimBLE v1.4 (and use MicroPython fork).

jimmo opened the pull request:
We need to implement a few features in NimBLE (that will hopefully eventually be upstreamed) such as host-side private addressing (for PYBD) and GATT database hashing. A new fork of NimBLE has been created in the micropython org, and this PR updates the submodule to point at it.

While we're here, might as well update to the latest release (v1.4). There are no specific fixes or features of note for MicroPython, but if we're going to be adding features, we might as well do it on the latest version. We have two additional commits to make it work in MicroPython, see https://github.com/apache/mynewt-nimble/pull/998 for background.

The changes to modbluetooth_nimble.c are because the registration for the secret store has been simplified.

jimmo committed:
extmod/nimble: Update to NimBLE v1.4.

We're using the MicroPython fork of NimBLE, which on the
`micropython_1_4_0` branch re-adds support for 64-bit targets and fixes
initialisation of g_msys_pool_list.

Also updates modbluetooth_nimble.c to suit v1.4.

Signed-off-by: Jim Mussared <jim.mussared@gmail.com>

jimmo committed:
extmod/nimble: Remove workaround for OS_ENOMEM.

This was fixed in NimBLE 1.4.

Signed-off-by: Jim Mussared <jim.mussared@gmail.com>

andrewleech commented:
Thanks for this @jimmo, however I think the update to the submodule url is missed in this PR?
```
diff --git a/.gitmodules b/.gitmodules
index 9fd45dfec..474ae9890 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -28,7 +28,7 @@
        url = https://github.com/hathach/tinyusb
 [submodule "lib/mynewt-nimble"]
        path = lib/mynewt-nimble
-       url = https://github.com/apache/mynewt-nimble.git
+       url = https://github.com/micropython/mynewt-nimble.git
 [submodule "lib/btstack"]
        path = lib/btstack
        url = https://github.com/bluekitchen/btstack.git
```

jimmo committed:
lib/mynewt-nimble: Switch to the MicroPython fork of NimBLE.

We will use this fork for adding further features and patches to support
MicroPython.

Signed-off-by: Jim Mussared <jim.mussared@gmail.com>
